### DISCUSSION TITLE
iOS libpag4.3.51崩溃

### DISCUSSION THREAD

**Discussion by JanyGee** [Tencent/libpag#2300] [General]
iOS系统：16.7.5 (20H307)、16.7.5 (20H307)、16.3.1 (20D67)、15.8.2 (19H384)、16.7.7 (20H330)都出现了崩溃
苹果崩溃日志
Thread 30 Crashed:
0   libsystem_kernel.dylib        	0x00000001bbb47bbc __pthread_kill + 8 (:-1)
1   libsystem_pthread.dylib       	0x00000001dc5e8844 pthread_kill + 208 (pthread.c:1668)
2   libsystem_c.dylib             	0x000000018b9a10b0 __abort + 124 (abort.c:155)
3   libsystem_c.dylib             	0x000000018b94c6b8 abort + 136 (abort.c:126)
4   libGLProgrammability.dylib    	0x00000001c2400600 glpCompiledShaderDeserialize + 100 (glp_linker.c:212)
5   libGLProgrammability.dylib    	0x00000001c2422480 ShCompile + 312 (glp_sh_interface.c:504)
6   GLEngine                      	0x00000001c2300620 gleShaderParse + 420 (gle_shader_parse.c:124)
7   GLEngine                      	0x00000001c22bc92c glCompileShaderIncludeARB_Exec + 468 (gl_shader_include.c:626)
8   libpag                        	0x00000001079c04c4 0x1078a0000 + 1180868
9   libpag                        	0x00000001079c034c 0x1078a0000 + 1180492
10  libpag                        	0x00000001079bb8c8 0x1078a0000 + 1161416
11  libpag                        	0x00000001079bb7fc 0x1078a0000 + 1161212
12  libpag                        	0x0000000107995900 0x1078a0000 + 1005824
13  libpag                        	0x00000001079bc760 0x1078a0000 + 1165152
14  libpag                        	0x000000010799987c 0x1078a0000 + 1022076
15  libpag                        	0x00000001079a3490 0x1078a0000 + 1062032
16  libpag                        	0x00000001079adc64 0x1078a0000 + 1104996
17  libpag                        	0x0000000107990520 0x1078a0000 + 984352
18  libpag                        	0x000000010798f160 0x1078a0000 + 979296
19  libpag                        	0x000000010799e00c 0x1078a0000 + 1040396
20  libpag                        	0x000000010791c814 pag::PAGSurface::draw(pag::RenderCache*, std::__1::shared_ptr<pag::Graphic>, pag::BackendSemaphore*, bool) + 648
21  libpag                        	0x000000010791a544 pag::PAGPlayer::flushInternal(pag::BackendSemaphore*) + 148
22  libpag                        	0x000000010791a46c pag::PAGPlayer::flush() + 140
23  libpag                        	0x00000001078ffe6c 0x1078a0000 + 392812
24  libpag                        	0x0000000107914a20 0x1078a0000 + 477728
25  libpag                        	0x0000000107915684 0x1078a0000 + 480900
26  libpag                        	0x00000001079d4b50 0x1078a0000 + 1264464
27  libpag                        	0x00000001079d4ee4 0x1078a0000 + 1265380
28  libpag                        	0x00000001079d590c 0x1078a0000 + 1267980
29  libsystem_pthread.dylib       	0x00000001dc5de338 _pthread_start + 116 (pthread.c:891)
30  libsystem_pthread.dylib       	0x00000001dc5dc938 thread_start + 8 (:-1)

bugly上报的日志，应该是同一个崩溃问题
![企业微信截图_53cb6c38-26da-4708-b02d-b603ec25df4f](https://github.com/Tencent/libpag/assets/15334578/1e633e58-57f5-4301-932b-714e0fd65f3a)

麻烦大佬看一下

**Comment by kevingpqi123** [comment]
1、为了提高对接效率，后续反馈问题的时候建议堆栈都还原好，dysm 文件我们都有提供，第1个问题等提供还原好后的堆栈后我们再来看；
2、截图的堆栈可能有些问题，和 pag 代码中的实际调用无法准确对应，根据之前的对接情况，bugly 上报的堆栈并不完全都是准确的，有一段时间  bugly 自身存在一些 bug，建议 bugly 升级至稳定版本再来看这里的上报

**Comment by JanyGee** [comment]
![企业微信截图_98daed53-3c1d-4e43-a8c3-5fb5595e29b2](https://github.com/Tencent/libpag/assets/15334578/ed9ee178-e233-467d-94eb-53861a3629af)
那这个可以吗？

**Reply by kevingpqi123** [reply]
还不太够，堆栈信息中有 libpag 相关的符号，这些符号是可以通过 dsym 文件还原出具体的堆栈，dsym 文件可以在 https://github.com/Tencent/libpag/releases 找到

**Reply by JanyGee** [reply]
新版本没有上报崩溃了，感谢
